src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/JavaNCSSCheck.java

```diff
@@ -83,7 +92,73 @@
  *   &lt;property name="methodMaximum" value="40"/&gt;
  * &lt;/module&gt;
  * </pre>
+ * <p>Example:</p>
+ * <pre>
+ * public void test() {
+ *   System.out.println("Line 1");
+ *   // another 38 lines of code
+ *   System.out.println("Line 40") // OK
+ *   System.out.println("Line 41") // violation, the method crosses 40 lines of uncommented code
+ * }
+ * </pre>
+ * <p>
+ * To configure the check with 100 allowed non commenting lines for a class:
+ * </p>
+ * <pre>
+ * &lt;module name="JavaNCSS"&gt;
+ *   &lt;property name="classMaximum" value="100"/&gt;
+ * &lt;/module&gt;
+ * </pre>
+ * <p>Example:</p>
+ * <pre>
+ * public class Test {
+ *   public void test() {
+ *       System.out.println("Line 1");
+ *       // another 48 lines of code
+ *       System.out.println("Line 49");
+ *   }
  *
+ *   public void test1() {
+ *       System.out.println("Line 50"); // OK
+ *       // another 48 lines of code
+ *       System.out.println("Line 98"); // violation
+ *   }
+ * }
+ * </pre>
+ * <p>
+ * To configure the check with 200 allowed non commenting lines for a file:
+ * </p>
+ * <pre>
+ * &lt;module name="JavaNCSS"&gt;
+ *   &lt;property name="fileMaximum" value="200"/&gt;
+ * &lt;/module&gt;
+ * </pre>
+ * <p>Example:</p>
+ * <pre>
+ * public class Test1 {
+ *   public void test() {
+ *       System.out.println("Line 1");
+ *       // another 48 lines of code
+ *       System.out.println("Line 49");
+ *   }
+ *
+ *   public void test1() {
+ *       System.out.println("Line 50");
+ *       // another 48 lines of code
+ *       System.out.println("Line 98"); // OK
+ *   }
+ * }
+ *
+ * public class Test2 {
```

Comment:
such example wont compile because of 2 top level classes

Reply:
Fixed it.